src/sinks/util/encoding.rs

```diff
@@ -61,11 +61,23 @@ impl Encoder<Event> for (Transformer, crate::codecs::Encoder<()>) {
         encoder
             .serialize(event, &mut bytes)
             .map_err(|error| io::Error::new(io::ErrorKind::InvalidData, error))?;
-        writer.write_all(&bytes)?;
+        write_all(writer, &bytes)?;
         Ok(bytes.len())
     }
 }
 
+/// Write the buffer to the writer, emitting an internal event which complies with the
+/// instrumentation spec- as this necessitates both an Error and EventsDropped event.
+pub(crate) fn write_all(writer: &mut dyn io::Write, buf: &[u8]) -> io::Result<()> {
+    writer.write_all(buf).map_err(|error| {
+        emit!(EncoderWriteError {
+            error: &error,
+            count: 1
```

Comment:
Is this always `1`? It seems like it could drop the whole batch of events, above.

Reply:
Yes, good catch :+1: